### PR TITLE
Add staffeln ut ci

### PR DESCRIPTION
**Pull request by ricolin**
relate to https://github.com/vexxhost/staffeln/issues/110

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
```
Comment (mnaser) [inline .pre-commit-config.yaml:6]: Let's add a new line after this, we've always put the different repos on their own.

File: requirements.txt [hunk 2]
```diff
@@ -20,5 +20,4 @@ parse
 tooz # Apache-2.0
 sherlock>=0.4.1 # MIT
 kubernetes # Apache-2.0
-# email
-# smtplib
+pre-commit
```
Comment (mnaser) [inline requirements.txt:23]: This shouldn't be here, we should move this to `test-requirements.txt` -- we cannot have Staffeln depend on `pre-commit`.

File: staffeln/conductor/backup.py [hunk 3]
```diff
@@ -501,8 +497,12 @@ def collect_instance_retention_map(self):
 
         try:
             servers = self.openstacksdk.get_servers(all_projects=True)
-        except OpenstackHttpException:
-            LOG.warn(_("Failed to list servers for all projects."))
+        except OpenstackHttpException as ex:
+            servers = []
```
Comment (mnaser) [inline staffeln/conductor/backup.py:501]: This seems to be a functional change I think here.

File: zuul.d/project.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+- project:
+    check:
+      jobs:
+        - staffeln-linters
+        - staffeln-unit
```
Comment (mnaser) [inline zuul.d/project.yaml:5]: Is it me or I don't think this job is defined anywhere?

File: .pre-commit-config.yaml
Comment (mnaser) [inline .pre-commit-config.yaml:1]: Let's use Black like we do in Cluster API for Magnum, I suggest using that as a reference point for the pre commit config :)

**Review by mnaser — CHANGES_REQUESTED** [review]

File: zuul.d/project.yaml (hunk 4 above)
Comment (ricolin) [reply zuul.d/project.yaml:5]: it's defined in zuul.d/jobs.yaml https://github.com/vexxhost/staffeln/pull/129/files/aca56c701f2d2f2e4ee150f0979116f4ece811e3#diff-f6856e72c485122d1cb0ed2470e60db446808bd69ccb662517be5c46e4048c4bR6

File: staffeln/conductor/backup.py (hunk 3 above)
Comment (ricolin) [reply staffeln/conductor/backup.py:490]: it's a bug I found. That servers can be left not defined if OpenstackHttpException
As we not looking for raise any error but only warning. it's weird to allow it failed on servers not defined :)

File: zuul.d/project.yaml
Comment (mnaser) [inline zuul.d/project.yaml:1]: Please add it to gate jobs too.

**Review by mnaser — CHANGES_REQUESTED** [review]

**Review by mnaser — APPROVED** [review]

**Pull request merged**